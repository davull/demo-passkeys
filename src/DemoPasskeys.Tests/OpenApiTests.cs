using FluentAssertions;
using Snapshooter.NUnit;

namespace DemoPasskeys.Tests;

public class OpenApiTests : IntegrationTestBase
{
    [TestCase("/swagger/index.html")]
    [TestCase("/swagger/v1/swagger.json")]
    public async Task Should_Return_Ok(string path)
    {
        var response = await GetAsync(path);
        response.EnsureSuccessStatusCode();
    }

    [Test]
    public async Task Swagger_Should_Match_Snapshot()
    {
        var content = await GetStringAsync("/swagger/v1/swagger.json");
        content.Should().MatchSnapshot();
    }
}