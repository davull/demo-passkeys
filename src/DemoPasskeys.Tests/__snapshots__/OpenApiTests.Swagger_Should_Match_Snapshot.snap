{
  "openapi": "3.0.1",
  "info": {
    "title": "DemoPasskeys",
    "version": "1.0"
  },
  "paths": {
    "/api/passkeys/PublicKeyCredentialCreationOptions": {
      "get": {
        "tags": [
          "PasskeysApi"
        ],
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    },
    "/api/passkeys/PublicKeyCredentialRequestOptions": {
      "get": {
        "tags": [
          "PasskeysApi"
        ],
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    },
    "/api/passkeys/UserPasskey": {
      "post": {
        "tags": [
          "PasskeysApi"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/StorePasskeyModel"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/StorePasskeyModel"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/StorePasskeyModel"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    },
    "/api/passkeys/Verify": {
      "post": {
        "tags": [
          "PasskeysApi"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/VerifyPasskeyModel"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/VerifyPasskeyModel"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/VerifyPasskeyModel"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "StorePasskeyModel": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string",
            "nullable": true
          },
          "userId": {
            "type": "string",
            "nullable": true
          },
          "publicKey": {
            "type": "string",
            "nullable": true
          },
          "publicKeyAlgorithm": {
            "type": "integer",
            "format": "int32"
          }
        },
        "additionalProperties": false
      },
      "VerifyPasskeyModel": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string",
            "nullable": true
          },
          "authenticatorData": {
            "type": "string",
            "nullable": true
          },
          "clientDataJson": {
            "type": "string",
            "nullable": true
          },
          "signature": {
            "type": "string",
            "nullable": true
          },
          "userHandle": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      }
    }
  }
}
